eclair-core/src/main/scala/fr/acinq/eclair/payment/offer/OfferManager.scala

```diff
@@ -0,0 +1,331 @@
+                      case None => ()
+                    }
+                  case _ => ()
+                }
+            }
+            case None => ()
+          }
+          Behaviors.same
+        case ReceivePayment(replyTo, paymentHash, payload) =>
+          signedMetadataCodec.decode(payload.pathId.bits) match {
+            case Attempt.Successful(DecodeResult(signed, _)) =>
+              registeredOffers.get(signed.offerId) match {
+                case Some(RegisteredOffer(offer, nodeKey, _, handler)) =>
+                  signed.verify(nodeKey.publicKey) match {
+                    case Some(metadata) if Crypto.sha256(metadata.preimage) == paymentHash =>
+                      val child = context.spawnAnonymous(PaymentActor(nodeParams, replyTo, offer, metadata))
+                      handler ! HandlePayment(child, signed.offerId, metadata.extraData)
+                    case _ => replyTo ! MultiPartHandler.GetIncomingPaymentActor.NoPayment
+                  }
+                case _ =>
+                  replyTo ! MultiPartHandler.GetIncomingPaymentActor.NoPayment
+              }
+            case Attempt.Failure(_) =>
+              replyTo ! MultiPartHandler.GetIncomingPaymentActor.NoPayment
+          }
+          Behaviors.same
+      }
+    })
+  }
+
+  object InvoiceRequestActor {
+
+    sealed trait Command
+
+    object RequestInvoice extends Command
+
+    /**
+     * Sent by the offer handler. Causes an invoice to be created and sent to the requester.
+     * @param amount         Amount for the invoice (must be the same as the invoice request if it contained an amount).
+     * @param routes         Routes to use for the payment.
+     * @param data           Some data for the handler by the handler. It will be sent to the handler when a payment is attempted.
+     * @param additionalTlvs additional TLVs to add to the invoice.
+     * @param customTlvs     custom TLVs to add to the invoice.
+     */
+    case class ApproveRequest(amount: MilliSatoshi,
+                              routes: Seq[ReceivingRoute],
+                              data: ByteVector,
+                              additionalTlvs: Set[InvoiceTlv] = Set.empty,
+                              customTlvs: Set[GenericTlv] = Set.empty) extends Command
+
+    /**
+     * Sent by the offer handler to reject the request. For instance because stock has been exhausted.
+     */
+    object RejectRequest extends Command
+
+    private case class WrappedInvoice(invoice: Try[Bolt12Invoice]) extends Command
+
+    private case class WrappedOnionMessageResponse(response: Postman.OnionMessageResponse) extends Command
+
+    def apply(nodeParams: NodeParams,
+              invoiceRequest: InvoiceRequest,
+              offerHandler: ActorRef[HandleInvoiceRequest],
+              nodeKey: PrivateKey,
+              router: akka.actor.ActorRef,
+              replyPath: OnionMessages.Destination,
```

Comment:
Naming that field `replyPath` is a bit confusing here, at first I thought this was _our_ reply path but we're not expecting a reply...what about naming this `senderPath` or something similar to clear that confusion?